migrations/20190122135305_create_order_deposit_transactions.up.sql

```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS OrderDepositTransactions (
+	orderId int(11) UNSIGNED NOT NULL,
+    transactionId int(11) UNSIGNED NOT NULL,
+    isAsk tinyint(1) NOT NULL,
+    createdAt varchar(255) NOT NULL DEFAULT "0000-00-00T00:00:00+05:30",
+	PRIMARY KEY (orderId, transactionId, isAsk)
+);
```

Comment:
Though there might be a bug in GORM, but as pointed by @gauthamk97 it works on auto-incremental ``id``, to be on safe side we should include it.

Reply:
> Foreign key for transactionID pointing to the Transactions table

I don't think this is necessary. We're already past the point of using references for this table. As we can't have a reference for order.

Reply:
@gauthamk97 , I'm unsure about not telling gorm.
One way is If there's a deadlock, ignore using tx.Save and use Get followed by Create/Update. Another way is to just add a auto inc primary key.
Yet another way is to ignore gorm as you suggested. I _guess_ this should be safe as long as Primary key constraint is there on the DB Table.

Just unsure because it doesn't feel clean. If cleanliness is a worry, pick either of first or second. Otherwise go ahead with third.

Reply:
@PaiAkshay998 why can't have a reference for order? I'd suggest you keep the references, it'll prevent you from shooting yourself in the foot.

Reply:
true. didn't realize. both of us noobed.

Reply:
true. didn't realize. both of us noobed.